crypto/poseidon/codehash.go

```diff
@@ -0,0 +1,57 @@
+package poseidon
+
+import (
+	"math/big"
+
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/crypto"
+)
+
+const defaultPoseidonChunk = 3
+
+func CodeHash(code []byte) (h common.Hash) {
+	// special case for nil hash
+	if len(code) == 0 {
+		return crypto.Keccak256Hash(nil)
+	}
+
+	cap := int64(len(code))
+
+	// Step1: pad code with 0x0 (STOP) so len(code) % 16 == 0
+	// Step2: for every 16 byte, convert it into Fr, so we get a Fr array
+	var length int
+
+	if len(code)%16 == 0 {
+		length = len(code) / 16
+	} else {
+		length = len(code)/16 + 1
+	}
+
+	Frs := make([]*big.Int, length)
+	ii := 0
+
+	for ii < length-1 {
+		Frs[ii] = big.NewInt(0)
+		Frs[ii].SetBytes(code[ii*16 : (ii+1)*16])
+		ii++
+	}
+
+	Frs[ii] = big.NewInt(0)
+	bytes := make([]byte, 16)
+	copy(bytes, code[ii*16:])
+	Frs[ii].SetBytes(bytes)
+
+	// Step3: pad Fr array with 0 to even length.
+	// FIXME: no need, Hash would pad it with 0 automatically, and Hash
```

Comment:
Does this need to be fixed or not?

Reply:
@noel2004 what do you think, can this fixme be removed?

Reply:
Now the step3 can be done inside `HashWithCap` and it is not recommended to be repeated outside.

So to make things clear I advice remove this 'step 3' completely and change the following step as 3 and 4.

Reply:
Updated.